src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -560,11 +572,9 @@ DeferErrorIfUnsupportedSubqueryPushdown(Query *originalQuery,
 	else if (!RestrictionEquivalenceForPartitionKeys(plannerRestrictionContext))
 	{
 		return DeferredError(ERRCODE_FEATURE_NOT_SUPPORTED,
-							 "cannot pushdown the subquery since all relations are not "
-							 "joined using distribution keys",
-							 "Each relation should be joined with at least "
-							 "one another relation using distribution keys and "
-							 "equality operator.", NULL);
+							 "complex joins are only supported when all distributed tables are "
+							 "joined on their distribution columns with equal operator",
```

Comment:
could we remove the "with equal operator" part? We also support joins on operators like IN, and it seems grammatically incorrect.